parser/prism/Translator.cc

```diff
@@ -2394,7 +2394,15 @@ ast::ExpressionPtr Translator::desugar(pm_node_t *node) {
         case PM_CLASS_NODE: { // Class declarations, not including singleton class declarations (`class <<`)
             auto classNode = down_cast<pm_class_node>(node);
 
-            auto name = desugar(classNode->constant_path);
+            ast::ExpressionPtr name;
+            if (classNode->constant_path == nullptr ||
```

Comment:
I think this is only repeated twice, but could you still extract a shared function anyway? I imagine this will always want to be kept the same between the class and module cases